File: CS/ViewModel/DialogViewModel.cs
using DataAnnotationAttributes.Model;
using DevExpress.Xpf.Mvvm;
using System;
using System.Linq;
using System.Windows.Input;

namespace DXSample {
    public class DialogViewModel : ViewModelBase {
        protected ContactContext Context;

        Contact entity;
        public Contact Entity {
            get {
                return entity;
            }
            set {
                entity = value;
                RaisePropertyChanged(() => Entity);
            }
        }
        public ICommand SaveCommand { get; private set; }

        public DialogViewModel() {
            SaveCommand = new DelegateCommand(Save);
        }

        protected override void OnParameterChanged(object parameter) {
            base.OnParameterChanged(parameter);
            if (!(parameter is int))
                return;
            Context = new ContactContext();
            Entity = Context.Contacts.Where(x => x.Id == (int)parameter).FirstOrDefault();
        }
        public void Save() {
            Context.SaveChanges();
            Messenger.Default.Send(new EntityChanged<Contact>(Entity.Id));
        }
    }
}

File: VB/ViewModel/DialogViewModel.vb
Imports DataAnnotationAttributes.Model
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports System
Imports System.Linq

Namespace DXSample
    Public Class DialogViewModel
        Implements ISupportParameter

        Protected Context As ContactContext1

        Public Overridable Property Entity() As Contact
        Public Overridable Property Parameter() As Object Implements ISupportParameter.Parameter

        Protected ReadOnly Property CurrentWindowService() As ICurrentWindowService
            Get
                Return Me.GetService(Of ICurrentWindowService)()
            End Get
        End Property

        Protected Overridable Sub OnParameterChanged()
            If Not (TypeOf Parameter Is Integer) Then
                Return
            End If
            Context = New ContactContext1()
            Entity = Context.Contacts.Where(Function(x) x.Id = DirectCast(Parameter, Integer)).FirstOrDefault()
        End Sub
        Public Sub Save()
            Context.SaveChanges()
            Messenger.Default.Send(New EntityChanged(Of Contact)(Entity.Id))
        End Sub
        Public Sub SaveAndClose()
            Save()
            CurrentWindowService.Close()
        End Sub
    End Class
End Namespace